src/software/ai/hl/stp/play/kickoff_enemy_play_test.cpp

```diff
@@ -28,18 +32,59 @@ TEST_F(KickoffEnemyPlayTest, test_kickoff_enemy_play)
     setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::PREPARE_KICKOFF_THEM);
 
     std::vector<ValidationFunction> terminating_validation_functions = {
-        // This will keep the test running for 9.5 seconds to give everything enough
-        // time to settle into position and be observed with the FullSystemGUI
-        // TODO: Implement proper validation
-        // https://github.com/UBC-Thunderbots/Software/issues/1397
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            // TODO: Fix bug with robot three not shadowing the enemy kicker in
+            // kickoff_enemy_play
+            // https://github.com/UBC-Thunderbots/Software/issues/1945
+
+            // Friendly robots in position to shadow enemy robots
+            Rectangle robotOneShadowingRect(Point(0, 2.2), Point(-0.4, 1.8));
+            Rectangle robotFiveShadowingRect(Point(0, -2.2), Point(-0.4, -1.8));
+            // Rectangle robotThreeShadowingRect(Point(-0.49, 0.1), Point(-0.75,
+            // -0.1));
+
+            robotInPolygon(1, robotOneShadowingRect, world_ptr, yield);
+            robotInPolygon(5, robotFiveShadowingRect, world_ptr, yield);
+            // robotInPolygon(3, robotThreeShadowingRect, world_ptr);
+
+            auto robotsDefendingPosts = [](std::shared_ptr<World> world_ptr) -> bool {
+                // Positions taken from kickoff_enemy_play
+                Point robotTwoExpectedPos =
+                    Point(world_ptr->field().friendlyGoalpostPos().x() +
+                              world_ptr->field().defenseAreaXLength() +
+                              2 * ROBOT_MAX_RADIUS_METERS,
+                          world_ptr->field().defenseAreaYLength() / 2.0);
+                Point robotFourExpectedPos =
+                    Point(world_ptr->field().friendlyGoalpostNeg().x() +
+                              world_ptr->field().defenseAreaXLength() +
+                              2 * ROBOT_MAX_RADIUS_METERS,
+                          -world_ptr->field().defenseAreaYLength() / 2.0);
+
+                double tolerance = 0.05;
+
+                Circle robotTwoCircle(robotTwoExpectedPos, tolerance);
+                Circle robotFourCircle(robotFourExpectedPos, tolerance);
+
+                Point robotTwoPos =
+                    world_ptr->friendlyTeam().getRobotById(2).value().position();
+                Point robotFourPos =
+                    world_ptr->friendlyTeam().getRobotById(4).value().position();
+
+                return contains(robotTwoCircle, robotTwoPos) &&
+                       contains(robotFourCircle, robotFourPos);
+            };
+
+            while (!robotsDefendingPosts(world_ptr))
```

Comment:
Should we be that thorough in checking the robot position during enemy kickoff? I thought all that our robots had to do was be on the friendly half during kickoff? I was just thinking maybe in the future we change how we position our robots during an enemy kickoff and the test would fail?

Reply:
Good question! You are correct that if we change the position of our robots significantly in the initialization of the test, it will fail. However, I'm not sure if that's what we're testing for. In the case that we update the positions, we should probably update the test accordingly. 
It will also be the same case that if Shadow Enemy tactic is changed significantly, then the test could fail. And as a result I think we should update the test then.

My understanding based on the [explanation](https://github.com/UBC-Thunderbots/Software/blob/f27212ab73b0bb04c7000ae3e9eef6f98ec4e92d/src/software/ai/hl/stp/play/kickoff_enemy_play.cpp#L55) within the `kickoff_enemy_play.cpp` was that the roles of the two robots defending either of the friendly goal posts and the three robots shadowing is an important part of the play. So I thought that it should be tested that they are in the general positions expected. Though this may be also making the test more fragile...